src/views/partition-table/components/Row.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <tr>
+    <td>
+      <input
+        class="input is-size-7-mobile is-size-7-tablet"
+        type="text"
+        placeholder="Name"
+        maxlength="16"
+        v-model="sName"
+      />
+    </td>
+    <td class="w-md">
+      <v-select
+        :options="types"
+        v-model="sType"
+        placeholder="Type"
+        taggable
+        selectOnTab
+      />
+    </td>
+    <td class="w-md">
+      <v-select
+        :options="subtypes"
+        v-model="sSubType"
+        label="label"
+        placeholder="Sub Type"
+        taggable
+        selectOnTab
+      />
+    </td>
+    <td>
+      <input
+        class="input is-size-7-mobile is-size-7-tablet"
+        type="text"
+        placeholder="Offset"
+        v-model="sOffset"
+      />
+    </td>
+    <td>
+      <input
+        class="input is-size-7-mobile is-size-7-tablet"
+        type="text"
+        placeholder="Size"
+        v-model="sSize"
+      />
+    </td>
+    <td><input type="checkbox" v-model="sFlag" /></td>
+    <td>
+      <a class="delete" @click="del"></a>
+    </td>
+  </tr>
+</template>
+
+<script lang="ts">
+import "vue-select/dist/vue-select.css";
+import { Component, Emit, PropSync, Vue } from "vue-property-decorator";
+import { PartitionTable } from "../store";
+import vSelect from "vue-select";
+Vue.component("v-select", vSelect);
+
+@Component
+export default class Row extends Vue {
+  @PropSync("name") sName: String;
+  @PropSync("type") sType: String;
+  @PropSync("subtype") sSubType: String;
+  @PropSync("offset") sOffset: String;
+  @PropSync("size") sSize: String;
+  @PropSync("flag") sFlag: String;
+
+  public get subtypes(): String[] {
+    if (this.sType === "app") {
+      return [
+        "factory",
+        "ota_0",
+        "ota_1",
+        "ota_2",
+        "ota_3",
+        "ota_4",
+        "ota_5",
+        "ota_6",
+        "ota_7",
+        "ota_8",
+        "ota_9",
+        "ota_10",
+        "ota_11",
+        "ota_12",
+        "ota_13",
+        "ota_14",
+        "ota_15",
+        "test",
+      ];
+    } else if (this.sType === "data") {
+      return ["ota", "phy", "nvs", "nvs_keys"];
+    }
+    return [];
```

Comment:
in all the cases the item returned is shown as autocomplete drop-down but user can type anything and it will be allowed ...which will be validated on save (implementation pending)